apps/settings/test/unit/panels/screen_lock_passcode/screen_lock_passcode_test.js

```diff
@@ -111,10 +114,16 @@ suite('ScreenLockPasscode > ', function() {
     });
 
     suite('passcode is different', function() {
-      setup(function() {
-        screenLockPasscode._settings.passcode = '0123';
-        screenLockPasscode._passcodeBuffer = '0000';
-        screenLockPasscode._checkPasscode();
+      setup(function(done) {
+        // set passcode to 0123
+        var promise = PasscodeHelper.set('0123');
+        // enter passcode 0000
+        promise = promise.then(() => {
+          screenLockPasscode._passcodeBuffer = '0000';
+          // check if we get the error for a wrong passcode
+          return screenLockPasscode._checkPasscode();
+        });
+        promise.then(() => { done(); });
```

Comment:
the `done` function does not like parameters, e.g. `promise.then(done)` would fail because done gets the promise result.
That's why I am using the `promise.then(() => { done(); });` in all cases. Would that work for you? Should I mention this with a comment in the code?